tests/unit/test_ringcells_reports.py

```diff
@@ -0,0 +1,196 @@
+    n = Node(create_tmp_simulation_config_file)
+    n.load_targets()
+    n.create_cells()
+    n.enable_reports()
+    assert len(n.reports) == 0
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize(
+    "create_tmp_simulation_config_file",
+    [
+        {
+            "simconfig_fixture": "ringtest_baseconfig",
+            "extra_config": {
+                "target_simulator": "NEURON",
+                "inputs": {
+                    "Stimulus": {
+                        "module": "pulse",
+                        "input_type": "current_clamp",
+                        "delay": 5,
+                        "duration": 50,
+                        "node_set": "RingA",
+                        "represents_physical_electrode": True,
+                        "amp_start": 10,
+                        "width": 1,
+                        "frequency": 50,
+                    }
+                },
+                "reports": {
+                    "soma_v": {
+                        "type": "compartment",
+                        "cells": "Mosaic",
+                        "variable_name": "v",
+                        "sections": "soma",
+                        "dt": 0.1,
+                        "start_time": 0.0,
+                        "end_time": 50.0,
+                    },
+                    "compartment_i": {
+                        "type": "compartment",
+                        "cells": "Mosaic",
+                        "variable_name": "i_membrane",
+                        "sections": "all",
+                        "dt": 1,
+                        "start_time": 0.0,
+                        "end_time": 40.0,
+                    },
+                },
+            },
+        }
+    ],
+    indirect=True,
+)
+def test_neuorn_compartment_report(create_tmp_simulation_config_file):
+    from neurodamus import Neurodamus
+    from neurodamus.core import NeurodamusCore as Nd
+    from neurodamus.core.configuration import SimConfig
+
+    n = Neurodamus(create_tmp_simulation_config_file)
+    assert len(n.reports) == 2
+
+    # For unit tests, we don't build libsonatareport to create the standard sonata reports,
+    # instead we use custom functions to record and write report vectors in ASCII format,
+    # but currently only for comparment reports
+    reports_conf = {name: conf for name, conf in SimConfig.reports.items() if conf["Enabled"]}
+    ascii_recorders = {}
+    for rep_name, rep_conf in reports_conf.items():
+        rep_type = rep_conf["Type"]
+        if rep_type == "compartment":
+            ascii_recorders[rep_name] = record_compartment_report(rep_conf, n._target_manager)
+    Nd.finitialize()  # reinit for the recordings to be registered
+    n.run()
+
+    # Write ASCII reports
+    for rep_name, (recorder, tvec) in ascii_recorders.items():
+        ascii_report = Path(n._run_conf["OutputRoot"]) / (rep_name + ".txt")
+        write_ascii_report(ascii_report, recorder, tvec)
```

Comment:
can we devise also a test where there are no compartments? So we check the edge case.

Also, do you think it would be useful to have this test also run with 2 cores so that the problems that arise when 2 players try to write the same stuff are covered? Following this also what happens if there are 2 cores, one has all the neurons and they try to write the ascii reports?

Reply:
>can we devise also a test where there are no compartments? So we check the edge case.

What is that? A cell with 0 comparment? Soma is also compartment. 

> Also, do you think it would be useful to have this test also run with 2 cores so that the problems that arise when 2 players try to write the same stuff are covered? Following this also what happens if there are 2 cores, one has all the neurons and they try to write the ascii reports?

The ASCII reporter here is not designed for multi-processing, as we don't need that for the official neurodamus run. `libsonata-report` does, but we don't build it for the unit tests.

Reply:
ok, in this case the tests are not needed